src/Lecture4.hs

```diff
@@ -183,8 +233,28 @@ The 'Stats' data type has multiple fields. All these fields have
 instance for the 'Stats' type itself.
 -}
 
-instance Semigroup Stats where
+-- define a strict version of <> similar to $!
+(<>!) :: (Semigroup a) => a -> a -> a
+(<>!)  !a !b = let !x = a <> b in x
+infixr 6 <>!
```

Comment:
This operator is useful potentially but here you don't need it because you already have `StrictData` enabled so the result of `<>` will be evaluated anyway 🙂